addons/binding/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/discovery/SmartthingsDiscoveryService.java

```diff
@@ -0,0 +1,244 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smartthings.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.smartthings.SmartthingsBindingConstants;
+import org.openhab.binding.smartthings.handler.SmartthingsBridgeHandler;
+import org.openhab.binding.smartthings.internal.SmartthingsHttpClient;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Smartthings Discovery service
+ *
+ * @author Bob Raker - Initial contribution
+ *
+ */
+public class SmartthingsDiscoveryService extends AbstractDiscoveryService implements EventHandler {
+
+    private static final int SEARCH_TIME = 30;
+    private static final int INITIAL_DELAY = 5;
+    private static final int SCAN_INTERVAL = 180;
+
+    private final Pattern findIllegalChars = Pattern.compile("[^A-Za-z0-9_-]");
+
+    private Logger logger = LoggerFactory.getLogger(SmartthingsDiscoveryService.class);
+
+    private SmartthingsBridgeHandler bridgeHandler;
+
+    private Gson gson;
+
+    private SmartthingsHttpClient httpClient = null;
+
+    private SmartthingsScan scanningRunnable;
+
+    private ScheduledFuture<?> scanningJob;
+
+    public SmartthingsDiscoveryService(SmartthingsBridgeHandler bridgeHandler) {
+        super(SmartthingsBindingConstants.SUPPORTED_THING_TYPES_UIDS, SEARCH_TIME);
+        logger.debug("Initializing discovery service");
+        this.bridgeHandler = bridgeHandler;
+
+        // Get a Gson instance
+        gson = new Gson();
+
+        this.scanningRunnable = new SmartthingsScan(this);
+        if (bridgeHandler == null) {
+            logger.warn("No bridge handler for scan given");
+        }
+        this.activate(null);
+    }
+
+    public SmartthingsDiscoveryService() {
+        super(SmartthingsBindingConstants.SUPPORTED_THING_TYPES_UIDS, SEARCH_TIME);
+        logger.debug("Initializing discovery service with default constructor");
+    }
+
+    @Override
+    protected void activate(Map<String, Object> config) {
+        logger.debug("SmartthingsDiscoveryService.activate() called");
```

Comment:
Please use the OSGi console for checking if a service is running and don't implement activate/modifed/deactivate only for logging